src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,23 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int FIGURES_COUNT = 5;
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Drawable getRandomFigure() {
+        return switch (random.nextInt(FIGURES_COUNT)) {
+            case 0 -> new Circle(10, colorSupplier.getRandomColor());
+            case 1 -> new Square(5, colorSupplier.getRandomColor());
+            case 2 -> new Rectangle(3, 7, colorSupplier.getRandomColor());
+            case 3 -> new RightTriangle(5, 9, colorSupplier.getRandomColor());
+            default -> new IsoscelesTrapezoid(4, 6, 5, colorSupplier.getRandomColor());
```

Comment:
Creating a figure, don't pass expressions in the constructor. Instead, create separate variables and pass them on for better code readability. For example, before the switch statement, you can define variables for the dimensions and color, and then use those variables when creating the figures.